src/libpsl-native/src/setdate.cpp

```diff
@@ -15,22 +15,44 @@
 
 //! @brief SetDate sets the date and time on local computer.
 //!      You must be super-user to set the time.
+//!      See comment in setdate.h about the use of private_tm
 //!
 //! SetDate
 //!
 //! @retval 0 successfully set date
 //! @retval -1 if failure occurred.
 //!
-int32_t SetDate(struct tm* time)
+int32_t SetDate(struct private_tm* time)
 {
     errno = 0;
 
     // Select locale from environment
     setlocale(LC_ALL, "");
 
     struct timeval tv;
+    int32_t result = GetTimeVal(*time,tv);
+    if(result != 0)
+    {
+        return result;
+    }
+
+    return settimeofday(&tv, NULL);
```

Comment:
Why do we use settimeofday if we don't use second parameter (timezone)?
Could we use [stime](https://linux.die.net/man/2/stime)? In the case we have to adjust only one parameter `time_t`.

Second question do this parameter adjustements work correctly for ARM reverse byte order?

Reply:
There are known issues there this isn't working on ARM.  @dantraMSFT  mentioned this to me.

Feel free to file an issue to switch to stime, although the better fix is probably to include the timezone.  This fix is scoped to stop corrupting memory.

Reply:
Seems the timezone can be a issue too. We should look this in depth. Will push new Issue to discuss  better solution.

Reply:
Tracking issue #6962